async_simple/CMakeLists.txt

```diff
@@ -23,6 +23,9 @@ if(UTHREAD)
 endif()
 
 # If there is no Uthread, async_simple is a header only library 
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  target_compile_options(async_simple_header_only INTERFACE "-fcoroutines")
```

Comment:
GCC enabled coroutines by default after 11. We need to check the version here. A message may be helpful here too.

Reply:
add check for gcc and gcc version